From 2886a52e391b44e07f0b051d3ab10ace27ab4b17 Mon Sep 17 00:00:00 2001
From: Watson Xu <xuhuashan@gmail.com>
Date: Wed, 1 Mar 2017 22:27:59 +0800
Subject: [PATCH 08/10] enable VI brightness to adjust independently.

so that the brightness does not change according to the
contrast adjustment
---
 src/platform/hi3518v200mpp/himpp-video-viu.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/platform/hi3518v200mpp/himpp-video-viu.cpp b/src/platform/hi3518v200mpp/himpp-video-viu.cpp
index 6a2fe7b..447a883 100644
--- a/src/platform/hi3518v200mpp/himpp-video-viu.cpp
+++ b/src/platform/hi3518v200mpp/himpp-video-viu.cpp
@@ -150,6 +150,18 @@ bool HimppViDev::enableObject()
     if (!_video_isp.enable())
         goto err_free_isp;
 
+    VI_MOD_PARAM_S param;
+    if ((s32Ret = HI_MPI_VI_GetModParam(&param)) == HI_SUCCESS) {
+        param.bLumaExtendEn = HI_TRUE;
+        param.bContrastModeEn = HI_TRUE;
+        if ((s32Ret = HI_MPI_VI_SetModParam(&param)) != HI_SUCCESS) {
+            HIMPP_PRINT("HI_MPI_VI_SetModParam failed [%#x]\n", s32Ret);
+        }
+    }
+    else {
+        HIMPP_PRINT("HI_MPI_VI_SetModParam failed [%#x]\n", s32Ret);
+    }
+
     dev_attr = *_video_sensor;
     if ((s32Ret = HI_MPI_VI_SetDevAttr(_devid, dev_attr)) != HI_SUCCESS) {
         HIMPP_PRINT("HI_MPI_VI_SetDevAttr %d failed [%#x]\n", _devid, s32Ret);
-- 
2.9.3

